core/src/main/scala/kafka/server/BrokerServer.scala

```diff
@@ -320,6 +320,11 @@ class BrokerServer(
           lifecycleManager.propagateDirectoryFailure(directoryId, config.logDirFailureTimeoutMs)
       }
 
+      /**
+       * TODO: move this action queue to handle thread so we can simplify concurrency handling
+       */
+      val defaultActionQueue = new DelayedActionQueue
```

Comment:
Could we pass this into KafkaApis too? This way, KafkaApis can call `tryCompleteActions()` directly there instead of through ReplicaManager.